### PR TITLE
Add `GetDetailedStatus` to support polling LRO intermediate progress

### PR DESCRIPTION
**Pull request by ArthurMa1978**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/MySqlFlexibleServerNextLinkOperationImplementation.cs [hunk 1]
```diff
@@ -0,0 +1,683 @@
...
+using System.Net;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using Azure.ResourceManager.MySql.MySqlFlexibleServers;
+
+namespace Azure.Core
+{
+    internal class MySqlFlexibleServerNextLinkOperationImplementation : IOperation
```
Comment (live1206) [inline sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/MySqlFlexibleServerNextLinkOperationImplementation.cs:20]: add TODO of adding policy applying in Azure.Core.NextLinkOperationImplementation

File: sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/BackupAndExportRestOperations.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+using Azure.ResourceManager.MySql.FlexibleServers.Models;
+using Azure.ResourceManager.MySql.MySqlFlexibleServers;
+
+namespace Azure.ResourceManager.MySql.FlexibleServers
+{
+    internal partial class BackupAndExportRestOperations
+    {
+        internal HttpMessage CreateCreateRequest(string subscriptionId, string resourceGroupName, string serverName, MySqlFlexibleServerBackupAndExportContent content)
+        {
+            var context = new RequestContext();
```
Comment (live1206) [inline sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/BackupAndExportRestOperations.cs:16]: Add TODO to be able to apply policy for operation

File: sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/Models/OperationProgressResult.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.ResourceManager.MySql.FlexibleServers.Models
+{
+    /// <summary> Represents Operation Results API Response. </summary>
+    public partial class OperationProgressResult : OperationStatusResult
```
Comment (live1206) [inline sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/Models/OperationProgressResult.cs:13]: Can we reuse the existing https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Generated/Models/OperationStatusExtendedResult.cs instead?

Comment (ArthurMa1978) [reply sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/Models/OperationProgressResult.cs:13]: This is different that service will add discriminator for different operation.

File: sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/MySqlFlexibleServerNextLinkOperationImplementation.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/MySqlFlexibleServerNextLinkOperationImplementation.cs:22]: Added

File: sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/BackupAndExportRestOperations.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/mysql/Azure.ResourceManager.MySql/src/MySqlFlexibleServers/Custom/RestOperations/BackupAndExportRestOperations.cs:16]: Added

**Review by live1206 — APPROVED** [review]

**Pull request closed without merging**